### ISSUE TITLE
ClippingPlanes do not propperly respond to translate values

### ISSUE THREAD

**Issue by Chris-N-K** [napari/napari#4503]
## 🐛 Bug

If an image layer has a translate value other then (0,0,0),  a `ClippingPlane` added to the layers `experimental_clipping_planes` appears to not be placed at the expected position. Tested with plane position calculated by hand and `layer.data_to_world()`. If the Image layer's translate is set back to (0,0,0) the clipping still does not work. Thus, it seems the plane position is not based on world coordinates.

## To Reproduce

```Python
import napari
import numpy as np

viewer=napari.Viewer()
viewer.add_image(np.random.randint(1,100,size=(100,100,10)))
# shape layer plane for reference
viewer.add_shapes(np.array([
    viewer.layers['Image'].data_to_world((0,0,5)),
    viewer.layers['Image'].data_to_world((0,100,5)),
    viewer.layers['Image'].data_to_world((100,100,5)),
    viewer.layers['Image'].data_to_world((100,0,5)),
    ])
)

# working clipping
viewer.layers['Image'].experimental_clipping_planes=[
    dict(normal=(0,0,1), position=viewer.layers['Image'].data_to_world((0,0,5)))
]

# change image position
viewer.layers['Image'].translate=(10,10,10) 
# -> image no longer visibly clipped

# try to change clipping to new position
viewer.layers['Image'].experimental_clipping_planes=[
    dict(normal=(0,0,1), position=viewer.layers['Image'].data_to_world((0,0,5)))
]
# -> no visible clipping

# translate back to (0,0,0) if clipping plane was not affected by translate at all it should clip again
viewer.layers['Image'].translate=(0,0,0) 
# -> image vanishes completely 
# -> clipping plane was affected by translate but not in the same way as the image layer
```

## Expected behavior
The clipping plane should either be moved when changing translate of the parent layer or at least follow the same world coordinates as the image when positioning.

## Environment

napari: 0.4.15
Platform: Linux-5.14.0-1034-oem-x86_64-with-glibc2.31
System: Ubuntu 20.04.4 LTS
Python: 3.9.12 | packaged by conda-forge | (main, Mar 24 2022, 23:25:59) [GCC 10.3.0]
Qt: 5.12.9
PyQt5: 5.12.3
NumPy: 1.22.3
SciPy: 1.8.0
Dask: 2022.03.0
VisPy: 0.9.6

OpenGL:
- GL version: 4.6 (Compatibility Profile) Mesa 21.2.6
- MAX_TEXTURE_SIZE: 16384

Screens:
- screen 1: resolution 1920x1080, scale 1.0
- screen 2: resolution 1920x1080, scale 1.0

Plugins:
- console: 0.0.4
- napari-itk-io: 0.1.0
- scikit-image: 0.4.15
- svg: 0.1.5

**Comment by brisvag** [comment]
Thanks for the code snippet. There's definitely somethign wrong with how `Image` (and `Labels`) handles clipping planes; as far as I can tell, the other layers behave as intended (that is, the planes stay put in scene coordinates, even if you translate the layer). A simple way to see the difference is to use the example `examples/clipping_planes_interactive_.py`, and then translate the layers. If the translation is small (relative to the size of the layers), you would expect to see minimal change. Instead, the clipping plane of the image layer goes in a very different place.

```python
for l in viewer.layers:
    l.translate = 0, 0, -1
```

However, the actual plane is still attached, and can be moved around, albeit at an offset (saeemingly unrelated to the translation amount).

Knowing that `Image` and `Labels` are using a different logic compared to all the other layers, so this point to the logic in vispy's `Volume` being wrong. It's strange cause I made them in parallel with the `PlanesClipper` filter, but somehting must be different.

**Comment by brisvag** [comment]
Ok, found the problem. I made a PR over at vispy/vispy#2329; hopefully we can get it in the next release :)

**Comment by Chris-N-K** [comment]
Thank you for the exceptional fast response :) your fix at vispy looks nice. Guess I can continue to play with the clipping planes ^^

**Comment by brisvag** [comment]
Closed by #4778! Sorry for the wait @ch-n, you can finally stop using your manual patch :P
